lib/lbt/bundle/AutoSplitter.js

```diff
@@ -226,7 +228,27 @@ class AutoSplitter {
 				props.store(out, "");
 				return out.toString().length();
 				*/
-				return fileContent.toString("latin1").length;
+
+				// input encoding is read from project configuration, default is "ISO-8859-1"
+				// since AutoSplitter is also used when splitting non-project resources (e.g. dependencies)
+				// *.properties files should be escaped if encoding option is specified
+				const propertiesFileEncoding = resource.getProject()
+					&& resource.getProject().resources
+					&& resource.getProject().resources.configuration
+					&& resource.getProject().resources.configuration.propertiesFileEncoding;
+				if (propertiesFileEncoding) {
+					const encoding = nonAsciiEscaper.getEncodingFromAlias(propertiesFileEncoding);
+					await nonAsciiEscaper({
+						resources: [resource.resource],
+						options: {
+							encoding
+						}
+					});
+				}
+
+				const fileContent = await resource.buffer();
+
+				return fileContent.toString().length;
```

Comment:
Right now this will not do escaping if there is no project (e.g. when bundler is used with manually created FS adapters). And in this case the behavior changes to read the files with utf-8 encoding, not latin1.

I would prefer to also do the escaping in that scenario and use latin1, as that has been the behavior so far.

Reply:
done, moved to separate module